platform/packages/platform/src/reply.rs

```diff
@@ -93,16 +93,18 @@ where
     M::decode(message).map_err(Into::into)
 }
 
-fn decode<M>(reply: Reply) -> Result<M>
+fn decode_first_response<M>(reply: Reply) -> Result<M>
 where
     M: Message + Default,
 {
-    reply
+    let responses = reply
         .result
         .into_result()
         .map_err(Error::ReplyResultError)?
-        .data
-        .ok_or(Error::EmptyReply())
-        .map_err(Into::into)
-        .and_then(|ref data| decode_raw(data))
+        .msg_responses;
+    if responses.is_empty() {
+        Err(Error::EmptyReply())
+    } else {
+        decode_raw(responses[0].value.as_slice())
+    }
```

Comment:
I feel list the more idiomatic thing to do here is to use pattern matching with `[T]::first`. Even better with `let [response] = &*responses else { ... }` or it's `if let`/`match` equivalent.